### PR TITLE
#992 projects page revamped

### LINKED ISSUE #992: Project Overview - Populate Project Overview Page
User Story:

As a user I want a page of projects with just the projects relevant to me, so I can see info on them at a glance and have ready access to all of them without having to sift through the projects page.

Blocked By #991 

Acceptance Criteria:

image

- When a lead/head clicks on the 'Projects' button, direct them to a new projects page view
- This project view should be labeled as "Overview", with an alternate one that is labeled "All Projects" which redirects to the old projects view is currently on this page.
- The personalized projects page should have a breadcrumbs displaying Home/Projects/Personalized Projects in the top left corner underneath the tabs.
- Label the page in the top right as the Projects page
- The page should contain one section, which is a grid of project cards

- The grid will be a flex wrap grid
- Sort the grid in the following manner; first display favorited projects, then projects the user is a lead / manager of, then projects on their subteam
- When a project in the overview is favorited, it will be auto moved to the favorites section

### PR DESCRIPTION
**Pull request by nezamj03**
## Changes

Added 3 pages - favorites, lead by, and in team. Only leadership or better has this view

## Notes

u gotta make the cards have consistent width / height because some of them are shorter and some are longer

## Screenshots

Screen Shot 2023-04-20 at 10 59 20 PM

Screen Shot 2023-04-20 at 10 59 31 PM

Screen Shot 2023-04-23 at 4 28 14 AM
Screen Shot 2023-04-23 at 4 29 11 AM

## To Do

look at notes

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x ] All commits are tagged with the ticket number
- [ x] No linting errors / newline at end of file warnings
- [ x] All code follows repository-configured prettier formatting
- [ x] No merge conflicts
- [ x] All checks passing
- [ x] Screenshots of UI changes (see Screenshots section)
- [ x] Remove any non-applicable sections of this template
- [ x] Assign the PR to yourself
- [ x] No `yarn.lock` changes (unless dependencies have changed)
- [ x] Request reviewers & ping on Slack
- [ x] PR is linked to the ticket (fill in the closes line below)

Closes #992

### CODE REVIEW

File: src/frontend/src/pages/ProjectsPage/ProjectsFavorited.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { Grid, Typography } from '@mui/material';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+
+/**
+ * Cards of all projects this user has favorited
+ */
+const ProjectsFavorited: React.FC = () => {
+  const { isLoading, data, isError, error } = useAllProjects();
+  const auth = useAuth();
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsFavorited.tsx:18]: useCurrentUser

File: src/frontend/src/pages/ProjectsPage/ProjectsInTeam.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Grid, Typography } from '@mui/material';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+
+/**
+ * Cards of all projects that this user is in their team.
+ */
+const ProjectsInTeam: React.FC = () => {
+  const { isLoading, data, isError, error } = useAllProjects();
+  const auth = useAuth();
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsInTeam.tsx:18]: useCurrentUser

File: src/frontend/src/pages/ProjectsPage/ProjectsInTeam.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (isLoading || !data) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error?.message} />;
+
+  const inTeamProjects = data?.filter((project) =>
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsInTeam.tsx:23]: this shouldnt be optional

File: src/frontend/src/pages/ProjectsPage/ProjectsLeadOrManager.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { Grid, Typography } from '@mui/material';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+
+/**
+ * Cards of all projects that this user is Project Lead/Manager for
+ */
+const ProjectsLeadOrManager: React.FC = () => {
+  const { isLoading, data, isError, error } = useAllProjects();
+  const auth = useAuth();
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsLeadOrManager.tsx:18]: useCurrentUser

File: src/frontend/src/pages/ProjectsPage/ProjectsLeadOrManager.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (isLoading || !data) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error?.message} />;
+
+  const projectLeadOrManagerProjects = data?.filter(
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsLeadOrManager.tsx:23]: this shouldnt be optional

File: src/frontend/src/pages/ProjectsPage/ProjectsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+import { useAuth } from '../../hooks/auth.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import { isLeadership } from 'shared';
+
+/**
+ * Cards of all projects that this user is in their team.
+ */
+
+const ProjectsPage: React.FC = () => {
+  const auth = useAuth();
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsPage.tsx:22]: useCurrentUser

File: src/frontend/src/pages/ProjectsPage/ProjectsTable.tsx [hunk 7]
```diff
@@ -116,9 +115,7 @@ const ProjectsTable: React.FC = () => {
 
   const theme = useTheme();
   return (
-    <>
-      <PageTitle title={'Projects'} previousPages={[]} />
-
+    <Grid container xs={12}>
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectsPage/ProjectsTable.tsx:118]: why does this need a grid container for one thing

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
I think we want these to be seperate sections on the same page so only two tabs with overview and all and then on overview we have three sections of favorited, as lead, and my teams projects or something

File: src/frontend/src/pages/ProjectsPage/ProjectsPage.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  const noPermissionsView = (
+    <>
+      <PageTitle title={'Projects'} previousPages={[]} />
+      <ProjectsTable />
+    </>
+  );
+
+  if (!auth.user) return <LoadingIndicator />;
+
+  return <>{isLeadership(auth.user.role) ? permissionView : noPermissionsView}</>;
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ProjectsPage/ProjectsPage.tsx:101]: i know the ticket says to but i think we should definitely not restrict this by role, everyone should see this but for guests the default tab should be the table (i think members can/should have favorites too)

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
we're gonna want the tabs to be links too I think, see reid's pr for that

**Comment by nezamj03** [comment]
> we're gonna want the tabs to be links too I think, see reid's pr for that

defo gonna be a tmr issue

File: src/frontend/src/pages/ProjectsPage/ProjectsPage.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+            sx={{ '&:hover': { backgroundColor: 'transparent' } }}
+          >
+            Overview
+          </Button>
+          <Menu anchorEl={overviewSelection} open={open} onClose={handleClose}>
+            <MenuItem onClick={() => handleViewClick('favorite')}>Favorites</MenuItem>
+            {isLeadership(user.role) && <MenuItem onClick={() => handleViewClick('lead')}>Lead Or Mananage</MenuItem>}
+            <MenuItem onClick={() => handleViewClick('team')}>Your Team</MenuItem>
+          </Menu>
+        </Grid>
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectsPage/ProjectsPage.tsx:77]: Sorry, I think the ticket on this one was kind of unclear, we want all 3 of these sections to be displayed on the overview page like it is on the CR overview page that Sean just did

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by anthonybernardi — APPROVED** [review]
yurrrrrrr

**Pull request merged**